db/view/view.cc

```diff
@@ -318,6 +320,50 @@ static bool is_partition_key_empty(
     }
 }
 
+// Checks if the result matches the provided view filter
+class view_filter_checking_visitor {
+    const schema& _base;
+    const view_info& _view;
+    const std::vector<bytes>& _pk;
+    const std::vector<bytes>& _ck;
+
+    bool _matches_view_filter = true;
+public:
+    view_filter_checking_visitor(const schema& base, const view_info& view, const std::vector<bytes>& pk, const std::vector<bytes>& ck)
+        : _base(base)
+        , _view(view)
+        , _pk(pk)
+        , _ck(ck)
+    {}
+
+    void accept_new_partition(const partition_key& key, uint64_t row_count) {}
```

Comment:
The visitor interface is the most convenient way of accessing the underlying row data, but it's actually much more generic than what we need - including the fact that it also reads partition keys, etc - while we already know that we're going to run the visitor on a single row with well defined primary key. So, in short, this implementation is just a stub and the only sensible thing we could do here is to perhaps assert that the partition key is exactly the same as the one we passed in the constructor (which it is, because it comes from the same row). I chose not to assert here to avoid unnecessary panic checks in runtime, so I just left the function bodies empty.

And yes, I could omit passing the partition key in the constructor and instead read it here, but I prefer this visitor to be explicit about the fact that it currently only works for a single row and that its primary key is known at construction time.

Reply:
Hm, after hitting the "Comment" button I started having second thoughts about this interface. I'll give your suggestion a chance and use it in v3 if it works fine.

Reply:
It actually makes the visitor cleaner - thanks. I'll rerun all tests and push v3 with the suggestion.